MovieDB/MovieDB/Scenes/Dashboard/DashboardCoordinator.swift

```diff
@@ -1,24 +1,25 @@
 import UIKit
 
-protocol DashboardCoordinatorType {
+protocol DashboardCoordinatorType: Coordinator {
     func goToWatchList()
     func goToMovieDetail(movieId: Int)
     func goToSearch(query: String)
 }
 
-final class DashboardCoordinator: Coordinator, DashboardCoordinatorType {
+final class DashboardCoordinator: DashboardCoordinatorType {
     var navigationController: UINavigationController?
 
     init(navigationController: UINavigationController?) {
         self.navigationController = navigationController
     }
 
     func start() {
-        var dashboardVC = DashboardViewController.instantiate()
-        let useCase = DashboardUseCase()
-        let viewModel = DashboardViewModel(coordinator: self,
-                                           useCase: useCase)
-        dashboardVC.bind(to: viewModel)
+        let viewModel = DashboardViewModel(useCase: DashboardUseCase())
+        var dashboardVC = DashboardViewController.instantiate().then {
+            $0.coordinator = self
+        }
+        dashboardVC.bind(viewModel: viewModel)
```

Comment:
can move it inside block code of then above?

Reply:
It cannot. In that block, the $0 is immutable. Also, the func bind is mutating (that's why I set var dashboardVC)